metadata-ingestion/src/datahub/api/entities/external/external_entities.py

```diff
@@ -0,0 +1,239 @@
+import logging
+from abc import abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Iterable, List, Optional, Union
+
+import cachetools
+from pydantic import BaseModel
+
+from datahub.api.entities.platformresource.platform_resource import (
+    PlatformResource,
+    PlatformResourceKey,
+)
+from datahub.ingestion.graph.client import DataHubGraph
+from datahub.metadata.urns import PlatformResourceUrn, Urn
+from datahub.utilities.search_utils import ElasticDocumentQuery
+
+logger = logging.getLogger(__name__)
+
+
+class PlatformResourceRepository:
+    def __init__(self, graph: DataHubGraph):
```

Comment:
Anyone coming across this class for maintenance or considering using it would appreciate a quick note on the caching strategy, especially why there's a limited maxsize and ttl and why current values should fit the purpose.